app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -196,7 +208,7 @@ function RawScatterPlot({
     if (scatterData && !isLoading) {
       processScatterPlot()
     }
```

Comment:
Consolidating the useEffects as you've done here makes me worry about performance.  Now we have to rebuild all the traces (which now takes a little longer for some plots thanks to the expression sorting) every time the graph gets resized.  We could include caching and if/else optimizations inside processScatterData, but I'd just as soon keep them separate, and have one effect for 'cosmetic change' and one for 'data change'

Reply:
Good point regarding performance. Decoupling as you (rightly) propose entails a non-trivial refactor of processScatterPlot. I've opened SCP-3865 to track that.